docs/rfcs/rfc-029-connect-in-update-mutations.md

```diff
@@ -0,0 +1,281 @@
+```
+
+This would merge an existing "HAS_SPONSOR" relation, overwriting the old relation with the properties provided in the `edge` input.
+
+### Questions
+
+* What should be the default behaviour? Existing MERGE behaviour? Or always create a new relationship?
+* How would this be presented in query results? One of each, or list all of the individual related nodes?
+* How would disconnect work? Disconnect one or disconnect all?
+* How do aggregations work? Do we aggregate across all relationships, or just one?
+
+## Proposed Solution
+
+* Add the possibility to create new connections, instead of always overwriting existing relationships.
+* The existing default behaviour should be kept the same for now to avoid a breaking change.
+* It should be possible to overwrite the default behaviour when specifying the `@relationship` directive. This can be achieved with a `defaultUpdateOperation` argument that accepts either CREATE or UPDATE. This should be defaulted to UPDATE to avoid breaking changes.
+* Provide the option to the user to make use of either behaviour at query time. This can be achieved with an `operation` argument on to the `connect` input that accepts either CREATE or UPDATE. If this argument is not provided, maintain the default behaviour defined by `defaultUpdateOperation`.
+* If there are several connections of the same type, to the same nodes, these should be represented with the nodes being returned multiple times in the response. This is already the current behaviour.
+* To maintain consistency with returning duplicate nodes if multiple relationships, aggregetions should be accross all relationships. This is already the current behaviour.
+* Disconnect should disconnect all relations that meet the query filters (e.g. could use limit 1 to make it delete only a single relationship). This is already the current behaviour.
+
+### Usage Examples
+
+#### `@relationship` definition
+
+Specifying the `defaultUpdateOperation` on the `@relationship` directive:
+
+```gql
+type Client {
+    id: String!
+    login: String!
+    sponsor: [Client!]! @relationship(type: "HAS_SPONSOR", properties: "HasSponsor", direction: OUT, defaultUpdateOperation: "CREATE")
+}
+
+interface HasSponsor @relationshipProperties {
+    type: String!
+    startDate: Date!
+    endDate: Date
+}
+```
+
+#### New update mutation
+
+Using the new `operation` argument:
+
+```gql
+mutation {
+    updateClients(
+        where: {
+            id: "1",
+        },
+        connect: {
+            operation: "CREATE", # The new operation input argument, that accepts either CREATE or UPDATE
+            sponsor: [
+                {
+                    where: {
+                        node: {
+                            id: "2"
+                        }
+                    },
+                    edge: {
+                        type: "newType2",
+                        startDate: "123"
+                    }
+                }
+            ]
+        }
```

Comment:
Right, some suggestions here. Where you currently have `operation` there would be no granularity for each connection within the array. So see the below example where I have moved `operation` into each element, and I also propose making it a boolean that I believe it will only ever have two values. I've made a suggestion below, but I'm open to alternatives here!

```suggestion
        connect: {
            sponsor: [
                {
                    where: {
                        node: {
                            id: "2"
                        }
                    },
                    edge: {
                        type: "newType2",
                        startDate: "123"
                    },
                    connectIfExists: true
                },
                {
                    where: {
                        node: {
                            id: "4"
                        }
                    },
                    edge: {
                        type: "newType2",
                        startDate: "123"
                    },
                    connectIfExists: false
                },
            ]
        }
```

When implementing this work, I would split it into two - firstly do this part, the Mutation arguments, and then move onto the directive argument, both merged in individual PRs.

Reply:
I definitely agree that the new position is better and that an enum probably wasn't necessary as I can't think of other behaviours that could be added in the future.

However, I don't know if I like `connectIfExists` as I feel like it's not clear that it would create duplicate connections. Could `newConnectionIfExists` work instead?